# awesome-biomedical-data-science-training
List of trianing resources for openly available, ideally freely available, data science training opportunities with a focus on biomedical applications. Contributions welcome...  

Inspired by [awesome-single-cell](https://github.com/seandavi/awesome-single-cell/blob/master/README.md), [awesome-microbes](https://github.com/stevetsa/awesome-microbes)...

[Andre Reindero Bioinformatics Jupyter Notebooks](https://gist.github.com/afrendeiro/8fa43f074a64590c67c30ec7b3141cb7) - A list of jupyter notebooks in bioinformatics

[Awesome-single-cell Tutorials and Workflows](https://github.com/seandavi/awesome-single-cell#tutorials-and-workflows) - Tutorials for single-cell analysis.

[Birmingham Insititute](https://www.bioinformatics.babraham.ac.uk/training.html) - Bioinformatics course, R, Python, Perl, etc.

[Common Workflow Language CWL](https://github.com/screx/cwl-tutorial) - Tutorials and [user guide](https://www.commonwl.org/user_guide/).

[Data Carpentry](https://datacarpentry.org/) - Data Carpentry develops and teaches workshops on the fundamental data skills needed to conduct research. Our mission is to provide researchers high-quality, domain-specific training covering the full lifecycle of data-driven research.  

[Data Science Central](https://www.datasciencecentral.com/) - Data Science Central is one of the leading repositories of Data Science content that is regularly updated with the latest trends across domains such as Artificial Intelligence, Machine Learning, Deep Learning, Analytics, Big Data, and much more.  

[Harvard University](https://online-learning.harvard.edu/subject/data-science) - Free online courses in R, visualization, data anlaysis, data wrangling, statistics, AI/ML.  

[Jeff Leek](https://github.com/jtleek) - Data science trianing in [R](https://github.com/jtleek/rpackages), [data sharing](https://github.com/jtleek/datasharing), [data analysis](https://github.com/jtleek/dataanalysis), [genomics paper](https://github.com/jtleek/genomicspapers). 

[National Institute of Health - Office of Data Science Strategy](https://datascience.nih.gov/news) - Find the latest news, announcements, and events related to data science at NIH here.  [Data Science Training at NIH](https://datascience.nih.gov/data-science-training-resources).  

[National Institutes of Health Library](https://www.nihlibrary.nih.gov/services/bioinformatics-support/online-bioinformatics-tutorials) - Training for a range of commercial bioinformatics resources and applications.

[Nextflow](https://nf-co.re/usage/nextflow) - Learn the basics of Nextflow and how to write your own Nextflow pipelines, including documentation, training, and workshops.

[Towards Data Science](https://towardsdatascience.com/) - Towards Data Science is a Medium publication brewing with audience-oriented content not just about Data Science, but a multitude of related technologies such as Machine Learning, Programming, Visualization, and Artificial Intelligence, spanning across more than 5800 published articles.  

[Workflow Description Lanugage](https://openwdl.org/)- Training and documentation for Workflow Description Langugage (WDL), which is a way to specify data processing workflows with a human-readable and -writeable syntax. 

# Best Practices


Noble, W. S. (2009). A quick guide to organizing computational biology projects. PLoS Computational Biology, 5(7), e1000424. http://doi.org/10.1371/journal.pcbi.1000424

[NIH Best Practices for Sharing Research Software](https://github.com/seandavi/awesome-single-cell#tutorials-and-workflows) - Best practices for sharing research software and source code, developed under research grants in any stage of development, in a free and open format.

[Ten great papers for biologists starting out in computational biology](https://widdowquinn.github.io/ten_great_papers/)

 - [Noble, W. S. (2009). A quick guide to organizing computational biology projects. PLoS Computational Biology, 5(7), e1000424.](http://doi.org/10.1371/journal.pcbi.1000424)

- [Sandve, G. K., Nekrutenko, A., Taylor, J., & Hovig, E. (2013). Ten simple rules for reproducible computational research. PLoS Computational Biology, 9(10), e1003285.](http://doi.org/10.1371/journal.pcbi.1003285)

- [Hart, E. M., Barmby, P., LeBauer, D., Michonneau, F., Mount, S., Mulrooney, P., et al. (2016). Ten Simple Rules for Digital Data Storage. PLoS Computational Biology, 12(10), e1005097.](http://doi.org/10.1371/journal.pcbi.1005097)

- [Schnell, S. (2015). Ten Simple Rules for a Computational Biologist’s Laboratory Notebook. PLoS Computational Biology, 11(9), e1004385.](http://doi.org/10.1371/journal.pcbi.1004385)

- [Kass, R. E., Caffo, B. S., Davidian, M., Meng, X.-L., Yu, B., & Reid, N. (2016). Ten Simple Rules for Effective Statistical Practice. PLoS Computational Biology, 12(6), e1004961](http://doi.org/10.1371/journal.pcbi.1004961)

- [Carey, M. A., Papin, J. A. (2018) Ten simple rules for biologists learning to program. PLOS Computational Biology 14(1): e1005871.](https://doi.org/10.1371/journal.pcbi.1005871)

- [Wilson, G., Aruliah, D. A., Brown, C. T., Chue Hong, N. P., Davis, M., Guy, R. T., et al. (2014). Best Practices for Scientific Computing,PLoS Biology 12(1), e1001745–2.](http://doi.org/10.1371/journal.pbio.1001745)

- [Blischak, J. D., Davenport, E. R., & Wilson, G. (2016). A Quick Introduction to Version Control with Git and GitHub. PLoS Computational Biology, 12(1), e1004668.](http://doi.org/10.1371/journal.pcbi.1004668)

- [Perez-Riverol, Y., Gatto, L., Wang, R., Sachsenberg, T., Uszkoreit, J, et al. (2016) Ten Simple Rules for Taking Advantage of Git and GitHub. PLOS Computational Biology 12(7): e1004947.](https://doi.org/10.1371/journal.pcbi.1004947)

- [Weinberger, C. J., Evans, J. A., & Allesina, S. (2015). Ten simple (empirical) rules for writing science. PLoS Computational Biology, 11(4), e1004205.](http://doi.org/10.1371/journal.pcbi.1004205)



